utils/das_data_sorter.py

```diff
@@ -1,76 +1,179 @@
+            return 0 
+
+    def average_data(self, data:pd.Series) -> pd.Series:
+        '''Returns a column of new data points that has been averaged, based on specified time unit. 
+
+        Finds the average of all data points within the same time interval. Appends this new value into an array, then continues
+        to the next time interval. Once all time intervals have been accounted for, returns the set of new data points.
+        '''
+        average_data_array = []
+
+        for index_array in self.indexes:
+            current_average = self.mean(data[index_array])
+            current_average = round(current_average, ndigits=2)
+            average_data_array.append(current_average)
+
+        return average_data_array
+    
+    # def gps_data(self, data:pd.Series) -> pd.Series:
+        '''currently doesn't work as intended'''
+        
+    #     new_gps_data = []
+    #     for index_array in self.indexes:
+    #         if 1 is in data[index_array]:
+    #             new_gps_data.append(1)
+    #         else:
+    #             new_gps_data.append(0)
+        
+    #     return new_gps_data
+
+    def location_data(self, data:pd.Series) -> pd.Series:
+        '''Handles the certain gps data and returns the data point at the last element of each time interval. 
+
+        gps_location requires a separate function because the values are strings, whereas the other data columns are 
+        integers/floats. Handles it so that it will always return the location data at the last element of each time interval.
+        '''
+        new_location_data = []
+        for index_array in self.indexes:
+            last_element_index = index_array[-1] 
+            last_element = data[last_element_index]
+            new_location_data.append(last_element)
+        
+        return new_location_data
+    
+    def write_to_output_file(self, file_output:str) -> None:
+        '''Creates new CSV file and writes new data onto CSV file.'''
+        final = pd.DataFrame({
+            "time": self.time,
+            "gps": self.gps,
+            "gps_location": self.gps_location,
+            "gps_course": self.gps_course,
+            "gps_speed": self.gps_speed,
+            "gps_satellites": self.gps_satellites,
+            "aX": self.ax,
+            "aY": self.ay,
+            "aZ": self.az,
+            "gX": self.gx,
+            "gY": self.gy,
+            "gZ": self.gz,
+            "thermoC": self.thermoc,
+            "tempF": self.thermof,
+            "pot": self.pot,
+            "cadence": self.cadence,
+            "power": self.power,
+            "reed_velocity": self.reed_velocity,
+            "reed_distance": self.reed_distance
+        })
+        final.to_csv(file_output, index=False)
+        print("Success! Output is written to {file_output}")
+
+if __name__ == '__main__':
+    data = pd.read_csv(args.input)
+    das_sort = DasSort(data, args.output, args.unit)
```

Comment:
Once you add the above changes. This will basically read as:
1. Load csv file
2. Sort csv file
3. Write csv file to a file

If a developer wants to know _how_ the data is sorted, they will read `DasSort` implementation. I hope this gives you an idea about why we would do this!